tests/components/smartthings/test_cover.py

```diff
@@ -0,0 +1,177 @@
+"""
+Test for the SmartThings cover platform.
+
+The only mocking required is of the underlying SmartThings API object so
+real HTTP calls are not initiated during testing.
+"""
+from pysmartthings import Attribute, Capability
+
+from homeassistant.components.cover import (
+    ATTR_CURRENT_POSITION, ATTR_POSITION, DOMAIN as COVER_DOMAIN,
+    SERVICE_CLOSE_COVER, SERVICE_OPEN_COVER, SERVICE_SET_COVER_POSITION,
+    STATE_CLOSING, STATE_OPENING)
+from homeassistant.components.smartthings import cover
+from homeassistant.components.smartthings.const import (
+    DOMAIN, SIGNAL_SMARTTHINGS_UPDATE)
+from homeassistant.const import ATTR_BATTERY_LEVEL, ATTR_ENTITY_ID
+from homeassistant.helpers.dispatcher import async_dispatcher_send
+
+from .conftest import setup_platform
+
+
+async def test_async_setup_platform():
+    """Test setup platform does nothing (it uses config entries)."""
+    await cover.async_setup_platform(None, None, None)
+
+
+async def test_entity_and_device_attributes(hass, device_factory):
+    """Test the attributes of the entity are correct."""
+    # Arrange
+    device = device_factory('Garage', [Capability.garage_door_control],
+                            {Attribute.door: 'open'})
+    entity_registry = await hass.helpers.entity_registry.async_get_registry()
+    device_registry = await hass.helpers.device_registry.async_get_registry()
+    # Act
+    await setup_platform(hass, COVER_DOMAIN, device)
+    # Assert
+    entry = entity_registry.async_get('cover.garage')
+    assert entry
+    assert entry.unique_id == device.device_id
+
+    entry = device_registry.async_get_device(
+        {(DOMAIN, device.device_id)}, [])
+    assert entry
+    assert entry.name == device.label
+    assert entry.model == device.device_type_name
+    assert entry.manufacturer == 'Unavailable'
+
+
+async def test_open(hass, device_factory):
+    """Test the cover opens doors, garages, and shades successfully."""
+    # Arrange
+    devices = {
+        device_factory('Door', [Capability.door_control],
+                       {Attribute.door: 'closed'}),
+        device_factory('Garage', [Capability.garage_door_control],
+                       {Attribute.door: 'closed'}),
+        device_factory('Shade', [Capability.window_shade],
+                       {Attribute.window_shade: 'closed'})
+    }
+    await setup_platform(hass, COVER_DOMAIN, *devices)
+    entity_ids = [
+        'cover.door',
+        'cover.garage',
+        'cover.shade'
+    ]
+    # Act
+    await hass.services.async_call(
+        COVER_DOMAIN, SERVICE_OPEN_COVER,
+        {ATTR_ENTITY_ID: entity_ids},
+        blocking=True)
+    # Assert
+    for entity_id in entity_ids:
+        state = hass.states.get(entity_id)
+        assert state is not None
+        assert state.state == STATE_OPENING
```

Comment:
When will the state become opened?

Reply:
Once the device reaches that position and sends the push update.

Reply:
(I'm not mocking that part out in this test) I'll modify `test_update_from_signal` so it demonstrates that.

Reply:
Are we testing that somewhere?

Reply:
`test_update_from_signal` demonstrates how it now gets from `OPENING` to `OPEN`.  I had a copy + paste carryover there too which I fixed.